pkg/updates/updates.go

```diff
@@ -0,0 +1,237 @@
+package updates
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/kardianos/osext"
+
+	getter "github.com/jesseduffield/go-getter"
+	"github.com/jesseduffield/lazygit/pkg/commands"
+	"github.com/jesseduffield/lazygit/pkg/config"
+	"github.com/sirupsen/logrus"
+)
+
+// Update checks for updates and does updates
+type Updater struct {
+	LastChecked string
+	Log         *logrus.Entry
+	Config      config.AppConfigurer
+	OSCommand   *commands.OSCommand
+}
+
+// Updater implements the check and update methods
+type Updaterer interface {
+	CheckForNewUpdate()
+	Update()
+}
+
+var (
+	projectUrl = "https://github.com/jesseduffield/lazygit"
+)
+
+// NewUpdater creates a new updater
+func NewUpdater(log *logrus.Logger, config config.AppConfigurer, osCommand *commands.OSCommand) (*Updater, error) {
+	contextLogger := log.WithField("context", "updates")
+
+	updater := &Updater{
+		LastChecked: "today",
+		Log:         contextLogger,
+		Config:      config,
+		OSCommand:   osCommand,
+	}
+	return updater, nil
+}
+
+func (u *Updater) getLatestVersionNumber() (string, error) {
+	time.Sleep(5)
```

Comment:
sleeping for 5 nanoseconds is probably a bug. Be explicit if it isn't: time.Sleep(5 * time.Nanosecond)